event/event.go

```diff
@@ -116,13 +149,55 @@ func (e Event) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	return nil
 }
 
+func extractEventType(r *http.Request) Type {
+	eventType := Type(r.Header.Get("event"))
+	if eventType == "" {
+		eventType = TypeHTTP
+	}
+	return eventType
+}
+
+func parseAsCloudEventBinary(event *Event, headers http.Header) *Event {
```

Comment:
I mentioned that in the previous review. It should not get Event as an argument. It should be similar to `parseAsCloudEvent`. So the arguments should be only what's needed which is headers and `body []byte`. Event should be created inside.